resolwe/permissions/tests/test_shortcuts.py

```diff
@@ -176,3 +188,435 @@ def test_user_permissions(self):
+        contributor = get_user_model().objects.create(username="contributor")
+        proc = Process.objects.create(name='Test process', contributor=contributor)
+        self.data = Data.objects.create(name='Test data', contributor=contributor, process=proc)
+        dummy_data = Data.objects.create(name='Dummy data', contributor=contributor, process=proc)
+
+        self.storage1 = Storage.objects.create(
+            name='Test storage',
+            json={},
+            data=self.data,
+            contributor=contributor,
+        )
+
+        self.storage2 = Storage.objects.create(
+            name='Test storage 2',
+            json={},
+            data=self.data,
+            contributor=contributor,
+        )
+
+        Storage.objects.create(
+            name='Dummy storage',
+            json={},
+            data=dummy_data,
+            contributor=contributor,
+        )
+
+        self.user = get_user_model().objects.create(username="test_user")
+        self.group = Group.objects.create(name="test_group")
+
+        self.storage_list_viewset = StorageViewSet.as_view(actions={
+            'get': 'list',
+        })
+
+        self.storage_detail_viewset = StorageViewSet.as_view(actions={
+            'get': 'retrieve',
+        })
+
+    def test_list_permissons(self):
+        request = factory.get('/', content_type='application/json')
+        force_authenticate(request, self.user)
+
+        resp = self.storage_list_viewset(request)
+        self.assertEqual(len(resp.data), 0)
+
+        assign_perm("view_data", self.user, self.data)
+        resp = self.storage_list_viewset(request)
+        six.assertCountEqual(self, [storage['id'] for storage in resp.data],
+                             [self.storage1.pk, self.storage2.pk])
+
+        remove_perm("view_data", self.user, self.data)
+        resp = self.storage_list_viewset(request)
+        self.assertEqual(len(resp.data), 0)
+
+    def test_detail_permissons(self):
+        request = factory.get('/', content_type='application/json')
+        force_authenticate(request, self.user)
+
+        resp = self.storage_detail_viewset(request, pk=self.storage1.pk)
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+
+        assign_perm("view_data", self.user, self.data)
+        resp = self.storage_detail_viewset(request, pk=self.storage1.pk)
+        self.assertEqual(resp.data['name'], 'Test storage')
+
+        remove_perm("view_data", self.user, self.data)
+        resp = self.storage_detail_viewset(request, pk=self.storage1.pk)
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_not_allowed_methods(self):
+        self.assertFalse(hasattr(StorageViewSet, 'update'))
+        self.assertFalse(hasattr(StorageViewSet, 'partial_update'))
+        self.assertFalse(hasattr(StorageViewSet, 'destroy'))
+        self.assertFalse(hasattr(StorageViewSet, 'create'))
+
+
+# tests copied from guardina.testapp.tests.test_shortcuts
```

Comment:
Yes, tests are the same (I've only deleted few of them that are difficult to run and are not relevant to us). The only difference is that different `get_objects_for_user` function is used, so their tests cannot be run directly.